### PR TITLE
Vulkan - Keep track of number of descriptors allocated from the pool

### PR DESCRIPTION
**Pull request by Jhonnyg**
This PR fixes a long running issue with incorrect tracking of number of descriptors created for the vulkan renderer.

### CODE REVIEW

File: engine/graphics/src/vulkan/graphics_vulkan_device.cpp [hunk 1]
```diff
@@ -74,36 +75,97 @@ namespace dmGraphics
...
+        {
+            allocator->m_DescriptorSetMax += allocator->m_DescriptorsPerPool;
+            AllocateDescriptorSets(allocator);
+        }
+
+    #ifdef _DEBUG
+        if (allocator->m_DescriptorsPerPool * allocator->m_DescriptorPools.Size() > 16768)
+        {
+            dmLogOnceWarning("Vulkan: There are more than 16768 descriptors (%d) in flight, this might be a performance issue.", m_DescriptorPools.Size());
+        }
```
Comment (Jhonnyg) [inline engine/graphics/src/vulkan/graphics_vulkan_device.cpp:130]: The limit is basically taken from thin air, but I think we want to be notified when someone reaches this value as there will be incentive to actually do some of these improvements in our vulkan renderer

File: engine/graphics/src/vulkan/graphics_vulkan_device.cpp [hunk 2]
```diff
@@ -74,36 +75,97 @@ namespace dmGraphics
...
+        DescriptorPool& pool = allocator->m_DescriptorPools[allocator->m_DescriptorPoolIndex];
+
+        if ((allocator->m_DescriptorsPerPool - pool.m_DescriptorCount) < num_descriptors)
+        {
+            allocator->m_DescriptorPoolIndex++;
+            if (allocator->m_DescriptorPoolIndex >= allocator->m_DescriptorPools.Size())
+            {
+                res = AllocateDescriptorPool(allocator, vk_device);
+            }
+        }
```
Comment (Jhonnyg) [inline engine/graphics/src/vulkan/graphics_vulkan_device.cpp:118]: We now keep adding pools if we need to (as suggested by samsungs documentation). Previously we resized the pool by removing the old pool and replacing it with a new pool, but according to samsung this is the preferred way.

File: engine/graphics/src/vulkan/graphics_vulkan_device.cpp (hunk 1 above)
Comment (JCash) [reply engine/graphics/src/vulkan/graphics_vulkan_device.cpp:130]: Yeah, it does look a bit curious, as it's not really related to any power of two (16384, 32768 etc)

**Review by JCash — DISMISSED** [review]

File: engine/graphics/src/vulkan/graphics_vulkan_device.cpp (hunk 1 above)
Comment (Jhonnyg) [reply engine/graphics/src/vulkan/graphics_vulkan_device.cpp:130]: Lol, I think I meant to use 16384 here.. 😶‍🌫️

**Pull request merged**